### PR TITLE
*: show cmd to check if all needed histograms are loaded

### PR DESCRIPTION
**Pull request by chrysan**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:
Histograms are loaded by need and async, which makes plans not stable. As the first step, provide a show cmd to check if all needed histograms are loaded successfully. This cmd can be used before statement in some cases.

### What is changed and how it works?
Provide a show syntax as "show histograms_in_flight" and returns the number of neededHistograms in flight.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [x] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by winoros — COMMENTED** [review]
The usage of it would be the cases like
```
show histograms_in_flight;
explain analyze SQL_TEXT;
maybe some wait
show histograms_in_flight;
explain analyze SQL_TEXT;
```
See the difference in the execution plan of the two?

**Comment by chrysan** [comment]
> The usage of it would be the cases like
> 
> ```
> show histograms_in_flight;
> explain analyze SQL_TEXT;
> maybe some wait
> show histograms_in_flight;
> explain analyze SQL_TEXT;
> ```
> 
> See the difference in the execution plan of the two?

explain analyze SQL_TEXT;
show histograms_in_flight until 0;
explain analyze SQL_TEXT;  => this time we get the correct final plan

Provide this cmd in case we don't know when the histograms are all loaded and we can get the correct final plan.

File: statistics/table.go [hunk 1]
```diff
@@ -259,6 +259,10 @@ func (n *neededColumnMap) Delete(col tableColumnID) {
 	n.m.Unlock()
 }
 
+func (n *neededColumnMap) Length() int {
+	return len(n.cols)
```
Comment (time-and-fate) [inline statistics/table.go:263]: I think we need an `RLock()` around this.

Comment (chrysan) [reply statistics/table.go:263]: Yes you are right. For some languages it's OK but for golang it's data race. https://stackoverflow.com/questions/52710882/is-len-thread-safe-in-golang
Change it to RWMutex BTW.

**Review by xuyifangreeneyes — COMMENTED** [review]
Do we need to provide `where table_id = ?` for the command? Maybe there are lots of tables in database and we only care about the tables related to the sql and want to exclude async loading of other tables' columns.

**Comment by chrysan** [comment]
> Do we need to provide `where table_id = ?` for the command? Maybe there are lots of tables in database and we only care about the tables related to the sql and want to exclude async loading of other tables' columns.

We can add when we see the real requirement.

**Review by xuyifangreeneyes — APPROVED** [review]
LGTM

**Review by winoros — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**